### PR TITLE
Update docs to use docker image

### PR DESCRIPTION
**Pull request by tephenavies**
This PR is a major update of getting started and deployment docs for docs.terria.io. Docs now emphasise using docker image & Kubernetes. Preview at https://docker-docs--terriajs-docs-v8.netlify.app/guide/

Test against TerriaJS/TerriaMap#644

### CODE REVIEW

**Comment by AnaBelgun** [comment]
Update 15 Feb 2023: 
- TODO - rewrite the deployment section

File: doc/README.md [hunk 1]
```diff
@@ -1,30 +1,18 @@
 [TerriaJS](http://terria.io) is an open-source framework for web-based geospatial catalog explorers.
 
-The documentation on this site is for applications using TerriaJS version 8. If you are still maintaining an application using TerriaJS version 7 or lower go to [docs-v7.terria.io](https://docs-v7.terria.io).
+-   [Getting Started](getting-started.md): Quick start guide to building your first TerriaJS application.
+-   [Customizing](customizing/README.md): Configure and tweak a TerriaJS application, including skinning and setting up the catalog.
+-   [Connecting to Data](connecting-to-data/README.md): Connect TerriaJS to your servers and data.
+-   [Deploying](deploying/README.md): Deploy a TerriaJS application in simple and advanced scenarios.
+-   [Contributing](contributing/README.md): Add new features to TerriaJS, be part of the TerriaJS development team, set up a development environment, write tests, and perform code reviews.
```
Comment (tephenavies) [inline doc/README.md:7]: Prettier reformats lists with extra spaces after bullets when using indent_size=4. It's a little ugly, but indent_size=4 is needed for mkdocs, and it's still better to use Prettier than to have to enforce indent_size=4 manually.

File: doc/customizing/cloning-and-building.md
Comment (tephenavies) [inline doc/customizing/cloning-and-building.md:1]: New doc file (basically the old getting started)

File: doc/deploying/README.md
Comment (tephenavies) [inline doc/deploying/README.md:1]: Reworded this one a bit to point to Docker & Kubernetes

File: doc/deploying/deploying-terriamap.md
Comment (tephenavies) [inline doc/deploying/deploying-terriamap.md:1]: Moved pm2 instructions to 1 subsection

File: doc/deploying/deploying-to-aws.md
Comment (tephenavies) [inline doc/deploying/deploying-to-aws.md:1]: Remove old aws deploy docs

File: doc/deploying/deploying-with-kubernetes.md
Comment (tephenavies) [inline doc/deploying/deploying-with-kubernetes.md:1]: Major simplification now that helm 3 and easier to use docker tooling (Docker Desktop & Rancher Desktop) exist.

File: doc/getting-started.md
Comment (tephenavies) [inline doc/getting-started.md:1]: Getting started overhaul based on Docker. Much of the previous content was moved to `customizing/cloning-and-building.md`.

File: doc/js/rewrite-links.js
Comment (tephenavies) [inline doc/js/rewrite-links.js:1]: This new link rewriter allows us to put links to code in our docs

File: gulpfile.js
Comment (tephenavies) [inline gulpfile.js:1]: Refactor tasks to allow running the quick doc generation stuff (CatalogMember page generation & mkdocs) without re-running `build-for-doc-generation`, which takes ages with Babel.

from:
```
[17:31:00] ├─┬ docs
[17:31:00] │ └─┬ <series>
[17:31:00] │   ├─┬ user-guide
[17:31:00] │   │ └─┬ <series>
[17:31:00] │   │   ├─┬ <parallel>
[17:31:00] │   │   │ ├── code-attribution
[17:31:00] │   │   │ └── build-for-doc-generation
[17:31:00] │   │   └── userGuide                  < Couldn't be run separately
[17:31:00] │   └── docs
```
to:

```
[17:26:15] ├─┬ docs
[17:26:15] │ └─┬ <series>
[17:26:15] │   ├─┬ <parallel>
[17:26:15] │   │ ├── code-attribution
[17:26:15] │   │ └── build-for-doc-generation
[17:26:15] │   ├─┬ render-guide                < This is now its own task
[17:26:15] │   │ └─┬ <series>
[17:26:15] │   │   ├── copyToBuild
[17:26:15] │   │   ├── generateMemberPages
[17:26:15] │   │   └── mkdocs
[17:26:15] │   └── docs
```

File: wwwroot/languages/en/translation.json
Comment (tephenavies) [inline wwwroot/languages/en/translation.json:1]: Couple of translation string improvements. These should probably have been a separate PR.

File: doc/contributing/model-layer.md [hunk 2]
```diff
@@ -194,18 +194,18 @@ This means that we can call `asyncLoader.load()` many times without worrying abo
...
 ```ts
 async function forceLoadX() {
-  const url = this.someObservableUrl;
-  const someData = await loadText(url);
-  runInAction(() => (this.someOtherObservable = someData));
+    const url = this.someObservableUrl;
+    const someData = await loadText(url);
+    runInAction(() => (this.someOtherObservable = someData));
 }
 ```
```
Comment (tephenavies) [inline doc/contributing/model-layer.md:210]: indent_size=4 is even worse for code blocks in md files, but there's no prettier or editorconfig configuration that would fix this, and using 2 spaces would require patching mkdocs as the author of the parser used believes that lists in lists must be indented by 4 spaces exactly to be valid markdown.

File: .editorconfig [hunk 3]
```diff
@@ -11,5 +11,9 @@ insert_final_newline = true
 [*.md]
 trim_trailing_whitespace = false
 
+[doc/**.md]
+# mkdocs requires 4 spaces
```
Comment (tephenavies) [inline .editorconfig]:
```suggestion
# mkdocs requires 4 spaces. With 2 space indents, nested lists do not work
```

**Comment by staffordsmith83** [comment]
UPDATE 21 June 2023:

Pete should be able to finish today after meeting

**Review by mwu2018 — APPROVED** [review]

**Pull request merged**